src/app/Main/Body/QueryPage/Results/TableResults/TableContent-RowWithColSums.component.tsx

```diff
@@ -63,12 +62,16 @@ export function TableContentRowWithColSumComponent(props: {
         </TableCell>,
     ];
 
+    let totalColSum = 0

```

Comment:
I wonder if this info also comes from the API? If so i would prefer the API given total sum.

Reply:
Ok, you mentioned the total number of isolates in your question about testing. So i would use the API given number, then testing here becomes obsolete.

Reply:
But then the individual sums come from the frontend and the total sum from the API. shouldn't I test then even more that both numbers match?

Reply:
Ok, you are right, then you should leave it as it is. The table component does the summation and it is a feature of this component. Or perhaps later of the ResultsContainer and a proper summation service as the summation could be also relevant for the Chart.
As for the test, it would be one crossing concerns (api and table view). What you could do or want to test here is summing the isolates after getting them from the api and check if the total sum is the right one, here you could throw an exception. As for the summation in the table component, here you would rely on that component and a runtime check would be overhead, e.g. you would also not check the outputs of any library. (That's not true for applications maintaining nuclear power plants!).
Instead you would check the algorithm with unit and e2e tests before you commit new code:
Unit Test: You could check if the row and col sums the table component calculates are the same or match the 100%.
E2E Test: You could check if the displayed total sum in the table is the same as the displayed total sum in the results info header.
As for the 100% you could just write that value to the cell.

Reply:
Okay. So I made an extra Ticket for testing?

Reply:
You could make an epic Ticket for e.g. writing e2e-Tests and add that specific test to list in the description, to collect things the tests should cover.
In general i would start to write e2e tests, because they cover more parts of your code at ones. But the integration of them into the deployment environment is a job for Dominic.